lib/circuitbox/circuit_breaker.rb

```diff
@@ -52,15 +52,17 @@ def run!(run_options = {})
         logger.debug "[CIRCUIT] closed: querying #{service}"
 
         begin
-          t1 = Time.now
+          start_time = Time.now.to_i
           response = if exceptions.include? Timeout::Error
             timeout_seconds = run_options.fetch(:timeout_seconds) { option_value(:timeout_seconds) }
             timeout (timeout_seconds) { yield }
           else
             yield
           end
-          t2 = Time.now
-          log_circuit_execution_time(t2 - t1)
+          end_time = Time.now.to_i
+          if circuit_options[:notify_circuit_execution_time]
```

Comment:
Going that route doesn't it make more sense to actually define an interface which would not depend on time at all?

```ruby
class Timed

  def initialize(notifier, metric_name)
     @notifier = notifier
     @metric_name = metrics_name
   end

  def run
     before = Time.now
     yield
     after = Time.now
     notifier.metric_gauge(metric_name, before - after)
  end
end
```

This way we don't have any of the timing logic in here but can easily provide a `NullObject` which removes all the calls to stats.
Now this could be used where we need it and the only thing we expect is